src/main/java/com/xeeva/catalog/CostCenterPage.java

```diff
@@ -1,532 +1,583 @@
+		String getReqByDateMessage = lblCartItemAddedMessage.getText();
+		TestReporter.logStep("Message after changing the RequiredBy date at CC_level : "+ getReqByDateMessage);
+		TestReporter.assertTrue(getReqByDateMessage.contains("date has been updated."), 
+				"Required by date has been updated at CC-Level.");
+
+	}
+
+	/**
+	 * @summary: Method to change the RequiredBy value at linelevel.
+	 * @author: Praveen Namburi, @version: Created 20-09-2016.
+	 */
+	public void change_RequiredByAtLineLevel(){
+		// after cart check out, application taking time to load cost center page 
+		lblCC.syncVisible(30, false);
+		String getReqByDate_beforeChange = getReqByDateAtLineLevel();
+		TestReporter.log("Get ReqByDate_before Change: " + getReqByDate_beforeChange);
+
+		driver.executeJavaScript("arguments[0].click();", dateCCLineLevel.get(0));
+		List<WebElement> nextMonthArrows = driver.findElements(By.xpath("html/body/div[@class='calendar']/"
+				+ "table/thead/tr[2]/td[4]"));
+		int getNextMonthArrowsSize = nextMonthArrows.size();
+		System.out.println("Next month arrows size: "+getNextMonthArrowsSize);
+		for(WebElement nextMonth : nextMonthArrows){
+			nextMonth.click();
+			List<WebElement> selectDates = driver.findElements(By.xpath("html/body/div[@class='calendar']/"
+					+ "table/tbody/tr[4]/td[@class='day']"));
+			for(WebElement selWeekDate : selectDates){
+				selWeekDate.click();
+				break;
+			}
+		}
+
+		//Verify the message -'The RequiredBy date has been updated' is displayed.
+		verify_RequiredByDateUpdated_atCCLevel();
+
+		List<WebElement> inputDates = dateCCInputLineLevel;
+		int getInputDatesSize = inputDates.size();
+		TestReporter.log("Total no. of input dates available at CC_line-level: " + getInputDatesSize);
+		//Iterate through each ReqBydate element and get text,
+		//verify the expected date with actual values.
+		for(WebElement inputDate : inputDates){
+			String getModifiedDate = inputDate.getAttribute("value");
+			TestReporter.log("Get ReqByDate after change: "+getModifiedDate);
+			TestReporter.assertNotEquals(getReqByDate_beforeChange, getModifiedDate, 
+					"Date has been updated at line-level for the id: - ["+inputDate.getAttribute("id")+"]");
+			break;
+		}
+
+	}
+
+	/**
+	 * @summary: Method to change the RequiredBy value at Header level.
+	 * @author: Praveen Namburi, @version: Created 22-09-2016.
+	 * @param daysOut
+	 */
+	public void change_RequiredByAtHeaderLevel(){
+		// after cart check out, application taking time to load cost center page 
+		lblCC.syncVisible(30, false);
+		//To click on calender at CC_HeaderLevel.
+		driver.executeJavaScript("arguments[0].click();", lnkDateCC_HeaderLvel);
+		List<WebElement> nextMonthArrows = driver.findElements(By.xpath("html/body/div[@class='calendar']/"
+				+ "table/thead/tr[2]/td[4]"));
+		int loopCount=0;
+		for(WebElement nextMonth : nextMonthArrows){
+			nextMonth.click();
+			driver.setElementTimeout(Constants.ELEMENT_TIMEOUT);
```

Comment:
So this line only sets the element timeout to the default (which it is already at).  If you need to change it from the default you will do something like this which will set it to 1 second -  'driver.setElementTimeout(1)' do the next lines of code, and then when finished, set it back to the default timeout - driver.setElementTimeout(Constants.ELEMENT_TIMEOUT);